File: mobile/lib/core/utils/helper/res_formatter.dart
class ResponseModel {
  final bool success;
  final int status;
  final String message;
  final dynamic data;

  ResponseModel({
    required this.success,
    required this.status,
    required this.message,
    required this.data,
  });

  factory ResponseModel.fromJson(Map<String, dynamic> json) => ResponseModel(
    success: json['success'],
    status: json['status'],
    message: json['message'],
    data: json['data'],
  );
}

File: backend/helper/res_formatter.go
package helper

type SuccessResponse struct {
	Success bool   `json:"success"`
	Status  int16  `json:"status"`
	Message string `json:"message"`
	Data    any    `json:"data"`
}

type FailedResponse struct {
	Success bool  `json:"success"`
	Status  int16 `json:"status"`
	Message any   `json:"message"`
}
